src/device/intrinsics.jl

```diff
@@ -331,10 +331,26 @@ macro cuStaticSharedMem(typ, dims)
     return esc(:(CUDAnative.generate_static_shmem(Val{$id}, $typ, Val{$dims})))
 end
 
+
+
 function emit_static_shmem{N}(id::Integer, jltyp::Type, shape::NTuple{N,Int})
-    if !haskey(llvmtypes, jltyp)
-        error("cuStaticSharedMem: unsupported type '$jltyp'")
+    var = Symbol(:@shmem, id)
+    len = prod(shape) * sizeof(jltyp)
+    align = Base.Threads.alignment(jltyp)
```

Comment:
We are asking the GC here what alignment it would use for the type. I don't recall if this is only correct for isbits or if it holds in general but it gives reasonable defaults.